### PR TITLE
Adds Features to WriteCOptions.

### PR DESCRIPTION
**Pull request by matthias-blume**
Eliminates now-redundant no_sandbox field.

(This makes treatment of the sandbox/no-sandbox feature uniform.)

### CODE REVIEW

File: include/wabt/c-writer.h [hunk 1]
```diff
@@ -18,15 +18,17 @@
...
 #include "wabt/common.h"
+#include "wabt/feature.h"
 
 namespace wabt {
 
 struct Module;
 class Stream;
 
 struct WriteCOptions {
+  WriteCOptions(const Features& features) : features(features) {}
```
Comment (dimitry-) [inline include/wabt/c-writer.h]: explicit

File: include/wabt/c-writer.h [hunk 2]
```diff
@@ -18,15 +18,17 @@
...
   std::string_view module_name;
-  bool no_sandbox;
+  const Features& features;
```
Comment (dimitry-) [inline include/wabt/c-writer.h:33]: Also maybe make a copy here? It is unclear from the outside that caller is responsible for keeping the instance alive. Alternatively pass a pointer to c-tor to make it clear(er) that the caller is owning this object.

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

File: include/wabt/c-writer.h (hunk 1 above)
Comment (matthias-blume) [reply include/wabt/c-writer.h]: done

File: include/wabt/c-writer.h (hunk 2 above)
Comment (matthias-blume) [reply include/wabt/c-writer.h:33]: Yeah, you are right.  Avoiding the copy is a needless micro-optimization.  Changed.

**Comment by matthias-blume** [comment]
Huh ?  There are now test failures which really shouldn't be there.  I'll look into it.

**Comment by matthias-blume** [comment]
I'm baffled by the test failures.  What am I missing?  This change should be a no-op...

**Pull request merged**